rust/extractor/src/translate/base.rs

```diff
@@ -45,12 +45,15 @@ macro_rules! post_emit {
         $self.extract_canonical_origin($node, $label.into());
     };
     (Struct, $self:ident, $node:ident, $label:ident) => {
+        $self.emit_derive_expansion($node, $label);
         $self.extract_canonical_origin($node, $label.into());
     };
```

Comment:
Would a single block like this work?
```
    (Adt, $self:ident, $node:ident, $label:ident) => {
        $self.emit_derive_expansion($node, $label);
    };
```

Reply:
no, because the macro needs to exactly match the concrete type to kick in.

If we had trait specializations we could whip something up like that.